src/server/routes/calendar.js

```diff
@@ -59,20 +59,20 @@ router.post('/addevent', function(req,res){
   if(!google_calendar) {
     var google_calendar = new gcal.GoogleCalendar(req.user.calAccessToken);
   }
-
+  console.log(req.body);
   // Parse JSON
-  var eventJSON = JSON.parse(req.body.eventInfo);
+  
   // Insert Event into Google Database
   // Logic to Associate Calendar Summary with ID
   google_calendar.calendarList.list(function(err, calendarList) {
 
     for(var i = 0; i < calendarList.items.length; i++){
-      if(eventJSON.calendar === calendarList.items[i].summary){
+      if(req.body.calendar === calendarList.items[i].summary){
```

Comment:
`if (req.body.calendar && req.body.calendar === calendarList.items[i].summary) {`

Also, else what? Throw an error, something. Everything wont always be as expected.